core/coreunix/add.go

```diff
@@ -153,8 +161,55 @@ func (params *Adder) PinRoot() error {
 	return params.node.Pinning.Flush()
 }
 
-func (params *Adder) Finalize(DAG dag.DAGService) (*dag.Node, error) {
-	return params.editor.Finalize(DAG)
+func (params *Adder) Finalize() (*dag.Node, error) {
+	root, err := params.mr.GetValue().GetNode()
+	if err != nil {
+		return nil, err
+	}
+
+	params.RootNode()
```

Comment:
i think i just forgot to remove that... good catch.  I try not to ever have weird side effects like that.

as an aside: I wish go would let you say "count unhandled error returns as compile errors"